src/plugins/vis_type_timeseries/public/application/components/lib/convert_series_to_datatable.ts

```diff
@@ -73,24 +95,49 @@ export const convertSeriesToDataTable = async (
         usedIndexPattern = indexPattern;
       }
     }
-    const isGroupedByTerms = layer.split_mode === 'terms';
+    const isGroupedByTerms = layer.split_mode === BUCKET_TYPES.TERMS;
+    const isGroupedByFilters = layer.split_mode === BUCKET_TYPES.FILTERS;
     const seriesPerLayer = series.filter((s) => s.seriesId === layer.id);
     let id = X_ACCESSOR_INDEX;
 
     const columns: TSVBColumns[] = [
-      { id, name: usedIndexPattern.timeFieldName || '', isSplit: false },
+      {
+        id,
+        name: usedIndexPattern.timeFieldName || '',
+        isMetric: false,
+        type: 'date_histogram',
+      },
     ];
     if (seriesPerLayer.length) {
       id++;
-      columns.push({ id, name: seriesPerLayer[0].splitByLabel, isSplit: false });
-      // Adds an extra column, if the layer is split by terms aggregation
+      columns.push({
+        id,
+        name: layer.metrics[0].field || seriesPerLayer[0].splitByLabel,
+        isMetric: true,
+        type: layer.metrics[0].type,
```

Comment:
Can you explain why this is necessary? As there can be multiple metrics and the first one might not even be used, I'm not sure whether it would work in all cases.

Reply:
Yes you are right, I changed it to the latest metric in the list. If I don't miss anything this is always the active one per layer on TSVB. 
https://github.com/elastic/kibana/pull/97426/commits/eb75702f27cd7652f6795cfadc5de553bcb41499